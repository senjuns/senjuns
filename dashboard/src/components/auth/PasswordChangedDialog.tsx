import { FC } from 'react';
import styled from 'styled-components';

import { ReactComponent as NotificationIcon } from '../../assets/svg/notification.svg';
import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogProps,
  Typography,
} from '../../components/common';

import { Sizes } from '../../shared/constants';

interface PasswordChangedDialogProps extends DialogProps {
  /**
   * Callback to be called when the dialog closes.
   */
  onClose: () => void;
}

export const PasswordChangedDialog: FC<PasswordChangedDialogProps> = ({
  open,
  onClose,
}) => {
  return (
    <StyledDialog open={open} onClose={onClose} aria-label="Password Changed">
      <Content>
        <NotificationIcon />
        <Typography variant="h6">Password changed!</Typography>
      </Content>
      <Actions>
        <Button color="secondary" fullWidth onClick={onClose}>
          Continue
        </Button>
      </Actions>
    </StyledDialog>
  );
};

const StyledDialog = styled(Dialog)`
  .MuiDialog-paper {
    border-radius: ${Sizes.xxLarge}px;
  }
`;

const Content = styled(DialogContent)`
  display: flex;
  align-items: center;
  flex-direction: column;
  gap: ${Sizes.small}px;
  margin-top: ${Sizes.xxLarge}px;
  padding: 0 ${Sizes.xxLarge}px;
`;

const Actions = styled(DialogActions)`
  display: flex;
  align-items: stretch;
  padding: ${Sizes.xxLarge}px;
`;
