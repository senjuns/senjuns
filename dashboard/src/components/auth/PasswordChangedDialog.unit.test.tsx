// eslint-disable-next-line import/no-extraneous-dependencies
import * as ShallowRenderer from 'react-test-renderer/shallow';
import { PasswordChangedDialog } from './PasswordChangedDialog';

describe('<PasswordChangedDialog />', () => {
  it('renders correctly.', async () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(<PasswordChangedDialog open={true} onClose={jest.fn()} />);

    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
