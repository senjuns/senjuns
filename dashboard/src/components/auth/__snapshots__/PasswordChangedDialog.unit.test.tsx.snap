// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordChangedDialog /> renders correctly. 1`] = `
<Styled(WithStyles(ForwardRef(Dialog)))
  aria-label="Password Changed"
  onClose={[MockFunction]}
  open={true}
>
  <Styled(WithStyles(ForwardRef(DialogContent)))>
    <ForwardRef(SvgNotification) />
    <Typography
      variant="h6"
    >
      Password changed!
    </Typography>
  </Styled(WithStyles(ForwardRef(DialogContent)))>
  <Styled(WithStyles(ForwardRef(DialogActions)))>
    <Button
      color="secondary"
      fullWidth={true}
      onClick={[MockFunction]}
    >
      Continue
    </Button>
  </Styled(WithStyles(ForwardRef(DialogActions)))>
</Styled(WithStyles(ForwardRef(Dialog)))>
`;
